### ISSUE TITLE
Make the FileAnnotationTab look great

### ISSUE THREAD

**Issue by LinusDietz** [JabRef/jabref#3109]
In https://github.com/JabRef/jabref/pull/3082 we have implemented the FileAnnotationTab in JavaFx and managed to create a working prototype.

Due to limited developing resources, however, it does not look very nice.
This issue is for somebody who wants to improve the design of the FileAnnotationTab.
![screen shot 2017-08-16 at 10 50 16](https://user-images.githubusercontent.com/1254003/29355327-ed07cbb8-8270-11e7-8f1e-b61459f63f1e.png)

**Comment by h93xV2** [comment]
I would like to solve this issue.

**Comment by muntrob** [comment]
Hi, I am new here and would like to take care of this issue.
Is it still available?
If yes, could someone please explain what the FileAnnotationTab should look like to be "great"?

**Comment by LinusDietz** [comment]
Hi @muntrob, it is, indeed!

Well, as you can see the interface does it's job, and has some buttons you can click on etc. However, with JavaFX these is the possibility to do way better. For example the labels Author, Page, Date look exactly like the concrete values (here dietzl, 1, etc), how about changing the layout of that?
Also if there are no annotations in a linked file, the left side (the table) stays blank, however, the overall structure does not change, despite nothing can be displayed on the right.

In a nutshell: We aren't designers, and most of us don't know much of UI-development, either. We are happy with any help to make JabRef look more modern, since this is also one factor that keeps researchers motivated doing literature work. (;

If you can help us, we would be very grateful and we'll support you by reviewing your Pull Request. A good strategy would be to make small changes, and open the PRs early, to be sure that the work goes into the right direction.

**Comment by muntrob** [comment]
Hi @LinusDietz,
after long silent here are my solution approaches.

**Comment by muntrob** [comment]
![no_annotations_variante_1](https://user-images.githubusercontent.com/33424784/33188701-35f07360-d09c-11e7-93b8-0b529e2941a8.jpg)
This is one way how the FileAnnotationTab could look like if there are no Annotations in the linked file.

**Comment by muntrob** [comment]
![no_annotations_variante_2](https://user-images.githubusercontent.com/33424784/33188787-fd8a6098-d09c-11e7-8e05-e67bce16320d.jpg)
Another, more simple way to show, that there are no annotations is just showing the message "File has no attached annotations" without displaying the attribute view on the right side. This solution keeps the structure simple and shows only the needed information.

**Comment by muntrob** [comment]
![with_annotations_new_layout](https://user-images.githubusercontent.com/33424784/33188858-cac147ca-d09d-11e7-81a4-6c623732a356.jpg)
And finally, if there are annotations, the FileAnnotationTab could contain a table which displays the attributes of the annotations much clearer and allows the list of annotations to be sorted by the attributes.

**Comment by LinusDietz** [comment]
Cool stuff!
Regarding the files with no annotations, I think I would prefer the grayed out version, as it does not break the overall design.

The new concept with the display of the marking and the content goes into to right direction. Actually I would only display one of those in the table to save a bit of space there.
Maybe its wise to move the copy button also in the table?

I'd suggest that you open a pull request soon, because this is usually the place where all developers discuss new features. It must not be perfect in the beginning, just indicate it's work in progress [WIP] and

**Comment by tobiasdiez** [comment]
I also like your redesign! Moving the some information to the left table and so concentrating on the content in the right pane is a very good idea!
The only criticism I have is the use of a table to show the overview of the annotations. Tables are usually great to compare or rank items, while lists in contrast are better to analyze or focus on a single item. Thus I would prefer a list-like approach here. Maybe something similar to [this here](http://www.designyourway.net/diverse/6/uilist/2655922.jpg) or the "message list" [here](http://www.material-ui.com/#/components/list): on the first line you display the author, page and a condensed version of the date; below this you give a preview of the annotation. What to you think?

If you guys prefer the table layout, I would suggest a lighter design similar to [here](https://blueyedgeek.github.io/mui-data-table/build/).

**Comment by lenhard** [comment]
The PR (or rather its replacement) has now been merged, so we can consider this issue as done.
